include/z64save.h

```diff
@@ -4,6 +4,32 @@
 #include "ultra64.h"
 #include "z64math.h"
 
+typedef enum MagicBarAction {
+    /* 00 */ MAGIC_BAR_ACTION_IDLE, // Regular gameplay
+    /* 01 */ MAGIC_BAR_ACTION_CONSUME_SETUP, // Sets the speed in which magic border flashes
+    /* 02 */ MAGIC_BAR_ACTION_CONSUME, // Consume magic until target is reached or no more magic is available
+    /* 03 */ MAGIC_BAR_ACTION_BORDER_CHANGE_1, // Flashes border and freezes Dark Link
+    /* 04 */ MAGIC_BAR_ACTION_BORDER_CHANGE_2, // Flashes border and draws yellow magic to preview target consumption
+    /* 05 */ MAGIC_BAR_ACTION_RESTORE_IDLE, // Reset colors and return to idle
+    /* 06 */ MAGIC_BAR_ACTION_BORDER_CHANGE_3, // Flashes border with no additional behaviour
+    /* 07 */ MAGIC_BAR_ACTION_LENS_CONSUME, // Magic slowly consumed by lens. 
+    /* 08 */ MAGIC_BAR_ACTION_GROW_WIDE, // Init magic on a new load, grow from a width of 0 to magicCapacity
+    /* 09 */ MAGIC_BAR_ACTION_FILL, // Add magic until full capacity is reached
+    /* 10 */ MAGIC_BAR_ACTION_ADD // Add requested magic
+} MagicBarAction;
+
+typedef enum MagicBarChange {
+    /* 00 */ MAGIC_BAR_CONSUME_NOW, // Consume Magic immediately without preview
+    /* 01 */ MAGIC_BAR_CONSUME_WAIT_NO_PREVIEW, // Sets consume target but waits to consume. No yellow magic preview to target consumption. Unused
+    /* 02 */ MAGIC_BAR_CONSUME_NOW_ALT, // Identical behaviour to MAGIC_BAR_CONSUME_NOW. Unused
+    /* 03 */ MAGIC_BAR_CONSUME_LENS, // Lens consumption
+    /* 04 */ MAGIC_BAR_CONSUME_WAIT_PREVIEW, // Sets consume target but waits to consume. Draws yellow magic to target consumption
+    /* 05 */ MAGIC_BAR_ADD // Sets a target to add magic
+} MagicBarChange;
```

Comment:
would call the enum ChangeType instead of just "Change". may need to update variables to reflect that.
also normally we would have a common prefix for these values that all fit under the same umbrella, but its a bit weird with `add` being different from all the other consume types i guess

in general I think we can probably afford to make all the magic related enums just start with `MAGIC_` and omit the bar? that goes for the enum above too.
Then we can do `MAGIC_CHANGE_CONSUME_NOW` and `MAGIC_CHANGE_ADD` etc, with `MAGIC` telling you its for the magic system, and `CHANGE` telling you its for the change type, followed by the actual type name

Reply:
For your first point:
I also had `CHANGE_` in the enum earlier, but then decided to take it out because the names were getting quite long and it seems like some of the names had reduncency in them i.e.
```
typedef enum MagicBarChange {
    /* 00 */ MAGIC_BAR_CHANGE_CONSUME_NOW,
    /* 01 */ MAGIC_BAR_CHANGE_CONSUME_WAIT_NO_PREVIEW,
    /* 02 */ MAGIC_BAR_CHANGE_CONSUME_NOW_ALT,
    /* 03 */ MAGIC_BAR_CHANGE_CONSUME_LENS,
    /* 04 */ MAGIC_BAR_CHANGE_CONSUME_WAIT_PREVIEW,
    /* 05 */ MAGIC_BAR_CHANGE_ADD
} MagicBarChange;
```
The terms `CHANGE_CONSUME` and `CHANGE_ADD` feel redundant, and there's only one that's different than the others `ADD` (I'm fine with `MagicBarChangeType` though)

For your second point, I messed around a bit with `magic` vs. `magicBar` for many of these names and enums. The reason I leaned more towards `MagicBar` when I could is because `magic` felt a bit too general, like magic spell or magic amount, it's hard to articulate. I felt `MagicBar` was really clear in what it was referring to, does that make sense?